nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/mail/Mailer.java

```diff
@@ -284,9 +271,7 @@ public void addInfoInMessageHeader(String address, AS as) throws MessagingExcept
                 } else {
                     replyToValue = new Address[oldValue.length + 1];
                 }
-                for (int i = 0; i < oldValue.length; i++) {
-                    replyToValue[i] = oldValue[i];
-                }
+                System.arraycopy(oldValue, 0, replyToValue, 0, oldValue.length);
```

Comment:
what if we replace `System.arraycopy` by `Address[] replyToValue = Arrays.copyOf(oldValue, oldValue.length, Address[].class);` i know that the `System.arraycopy` can be more performante then the `Arrays.copyOf` and `Arrays.copyOf` under hood uses  `System.arraycopy` after it creates the array for us. actually the call of this method is made in `org.nuxeo.ecm.automation.core.operations.notification.SendMail#addMailBoxInfoInMessageHeader` whicj iterate overs the persons address perhaps its the reason of using `System.arraycopy`

Reply:
I just used the sonar/idea replacement proposal.

Furthermore, I think your solution doesn't handle when `oldValue` is null (so does `Arrays.copyOf`).

Reply:
yep for sonar/idea ;)

for the second point from what i saw on the javadoc of the two methods, both of them will throw a `NPE` if the `src` array is null and for `System.arraycopy` will throw NPE if `dest` is null also.

Reply:
I reverted the change, this was working previously didn't want to introduce a regression on an old class.